tests/v2_validation/cattlevalidationtest/core/test_ldap_auth.py

```diff
@@ -209,6 +210,70 @@ def test_ldap_delete_token_on_logout(admin_client):
     assert identities.status_code == 401
 
 
+# 3
+@if_test_ldap
+def test_restrict_concurrent_sessions(admin_client):
+    # Use admin's token to enable restrict concurrent sessions
+    ldap_main_user = os.environ.get('LDAP_MAIN_USER')
+    ldap_main_pass = os.environ.get('LDAP_MAIN_PASS')
+
+    admin_token = get_authed_token(username=ldap_main_user,
+                                   password=ldap_main_pass)
+
+    cookies = dict(token=admin_token['jwt'])
+
+    update_setting_url = cattle_url() + \
+        '/settings/api.auth.restrict.concurrent.sessions'
+
+    data = {
+        "activeValue": "false",
+        "id": "api.auth.restrict.concurrent.sessions",
+        "name": "api.auth.restrict.concurrent.sessions",
+        "source": "Code Packaged Defaults",
+        "value": "true"
+    }
+
+    response = requests.put(update_setting_url, data=json.dumps(data),
+                            cookies=cookies)
+    assert response.ok
+
+    # Use user's token to test restrict
+    ldap_user2 = os.environ.get('LDAP_USER2')
+    ldap_pass2 = os.environ.get('LDAP_PASS2')
+
+    user_token_1 = get_authed_token(username=ldap_user2,
+                                    password=ldap_pass2)
+
+    cookies = dict(token=user_token_1['jwt'])
+    identities = requests.get(cattle_url() + "identities", cookies=cookies)
+    assert identities.status_code == 200
+
+    # Start another session
+    user_token_2 = get_authed_token(username=ldap_user2,
+                                    password=ldap_user2)
+
+    cookies = dict(token=user_token_2['jwt'])
+    identities = requests.get(cattle_url() + "identities", cookies=cookies)
+    assert identities.status_code == 200
+
+    # Token from first session shouldn't work now
+    cookies = dict(token=user_token_1['jwt'])
+    identities = requests.get(cattle_url() + "identities", cookies=cookies)
+    assert identities.status_code == 401
+
+    data = {
```

Comment:
The logic to flip it back really should be in a finalizer so that it runs regardless of whether the test passed